src/js/personalization/va-profile-beta/actions/index.js

```diff
@@ -71,28 +71,29 @@ function saveFieldHandler(apiRoute, requestStartAction, requestSuccessAction, re
 }
 
 async function sendProfileRequests() {
-  const result = {};
-  const requests = [
+  return [
     ['email', '/profile/email'],
     ['primaryTelephone', '/profile/primary_phone'],
     ['alternateTelephone', '/profile/alternate_phone'],
     ['mailingAddress', '/profile/mailing_address'],
     ['serviceHistory', '/profile/service_history'],
     ['personalInformation', '/profile/personal_information'],
     ['userFullName', '/profile/full_name']
-  ];
-
-  /* eslint-disable no-await-in-loop */
-  for (const [property, url] of requests) {
+  ].map(async ([propertyName, apiRoute]) => {
+    let propertyValue = null;
     try {
-      const response = await apiRequest(url);
-      result[property] = response.data.attributes;
+      const response = await apiRequest(apiRoute);
+      propertyValue = response.data.attributes;
     } catch (err) {
-      // Allow the property to remain undefined
+      // Let propertyValue remain null
     }
-  }
-
-  return result;
+    return [propertyName, propertyValue];
+  }).reduce(async (resultPromise, pendingRequest) => {
+    const result = await resultPromise;
+    const [propertyName, propertyValue] = await pendingRequest;
+    result[propertyName] = propertyValue;
+    return result;
+  }, {});
 }
```

Comment:
I'm really curious about thoughts on this. The .map above already sent all of requests, and this .reduce just constructs the profile result. I can't decide if it's too cryptic, or if as await/async become more mainstream this will become a common pattern for async functional code.

Reply:
Yeah, I was going back and forth. I don't think it's clearer than using `Promise.all`, but I'm not sure if that's just because I'm not used to async/await yet.

Reply:
maybe you could add a one-line comment that clarifies what this map/reduce is doing in case people reading this are not familiar with async/await

Reply:
Yeah my feelings exactly. I'm gonna merge this as-is but will revisit it. I think this code is neat but may not be worth it if it's overly complex.

Reply:
@bshyong whoops missed your comment. Good idea though, will do!